### PR TITLE
fix(api): Add NotFound exception on passing an invalid roleId while inviting user in workspace

### LINKED ISSUE #405: API: Error while inviting user to workspace
**Describe the bug**

When we try to invite a user to a workspace, and we try to assign them a role ID that doesn't exist, instead of returning a 404 with proper error message, the API throws a 500 error.

**To Reproduce**
1. Create a workspace
2. Try to invite another user to the workspace and in `roleIds` array, supply an ID that doesn't exist
3. Encounter the error

**Expected behavior**

If a workspace role provided in the `roleIds` array doesn't exist, we want to return a 404 specifying that the particular role doesn't exist.

**Screenshots**
- Postman request:
image
- Response:
image
- Server logs:
```
api:dev: [ERROR] 2024-07-30 14:21:38 - 
api:dev: Invalid `this.prisma.workspaceMember.create()` invocation in
api:dev: /home/raj/oss/keyshade.xyz/keyshade-xyz/apps/api/src/workspace/service/workspace.service.ts:983:60
api:dev: 
api:dev:   980 }
api:dev:   981 
api:dev:   982 // Create the workspace membership
api:dev: → 983 const createMembership = this.prisma.workspaceMember.create(
api:dev: An operation failed because it depends on one or more records that were required but not found. No 'WorkspaceRole' record(s) (needed to inline the relation on 'WorkspaceMemberRoleAssociation' record(s)) was found for a nested connect on one-to-many relation 'WorkspaceMemberRoleAssociationToWorkspaceRole'.
```

**Desktop (please complete the following information):**
 - OS: Ubuntu
 - Browser: Postman
 - Version: 2.3.0

**Additional context**

We also doubt that the same error will be present in `Update member role` endpoint aswell.

### PR DESCRIPTION
**Pull request by PriyobrotoKar**
## Description

Added validation to ensure that If a workspace role is provided in the `roleIds` array doesn't exist, a `NotFoundException` is thrown.

Fixes #405 

## Dependencies

No new dependencies were added.

## Future Improvements

_Mention any improvements to be done in future related to any file/feature_

## Mentions

_Mention and tag the people_

## Screenshots of relevant screens

_Add screenshots of relevant screens_

## Developer's checklist

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-check on my work

**If changes are made in the code:**

- [x] I have followed the [coding guidelines](https://google.github.io/styleguide/jsguide.html)
- [x] My changes in code generate no new warnings
- [ ] My changes are breaking another fix/feature of the project
- [ ] I have added test cases to show that my feature works
- [ ] I have added relevant screenshots in my PR
- [ ] There are no UI/UX issues

### Documentation Update

- [ ] This PR requires an update to the documentation at [docs.keyshade.xyz](https://docs.keyshade.xyz)
- [ ] I have made the necessary updates to the documentation, or no documentation changes are required.

### CODE REVIEW

File: apps/api/src/workspace/service/workspace.service.ts
Comment (rajdip-b) [inline apps/api/src/workspace/service/workspace.service.ts:1]: The code looks perfect! Just would need a little refactoring. Could you move out the code that you added into a function and then call the function instead?

**Review by rajdip-b — CHANGES_REQUESTED** [review]

File: apps/api/src/workspace/service/workspace.service.ts [hunk 1]
```diff
@@ -414,6 +414,26 @@ export class WorkspaceService {
...
+        id: {
+          in: roleIds
+        },
+        workspaceId: workspace.id
+      }
+    })
+
+    const invalidRoles = roleIds.filter(
+      (id) => !roles.map((role) => role.id).includes(id)
+    )
```
[the remark this answers is not in the document]
Comment (rajdip-b) [reply apps/api/src/workspace/service/workspace.service.ts:429]: @PriyobrotoKar also add this suggestion.

File: apps/api/src/workspace/service/workspace.service.ts
Comment (PriyobrotoKar) [reply apps/api/src/workspace/service/workspace.service.ts:1]: yes, right! Doing it now.

File: apps/api/src/workspace/service/workspace.service.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (PriyobrotoKar) [reply apps/api/src/workspace/service/workspace.service.ts:429]: okk 👍

**Comment by PriyobrotoKar** [comment]
@rajdip-b I have made those changes.

**Review by rajdip-b — APPROVED** [review]

**Pull request merged**